libraries/AP_Scripting/lua_scripts.cpp

```diff
@@ -37,6 +37,24 @@ lua_scripts::lua_scripts(const AP_Int32 &vm_steps, const AP_Int32 &heap_size, co
     _heap = hal.util->allocate_heap_memory(heap_size);
 }
 
+lua_scripts::~lua_scripts() {
+    repl_cleanup();
+
+    while (scripts != nullptr) {
+        remove_script(lua_state, scripts);
+    }
+
+    lua_close(lua_state);
+    hal.util->heap_realloc(_heap, lua_state, 0);
```

Comment:
You don't need to spend any time cleaning up the heap if you are going to free the whole heap. (It's technically correct, but it is just wasted cycles.